### DISCUSSION TITLE
Rosetta x86 install

### DISCUSSION THREAD

**Discussion by nagydavid** [pyenv/pyenv#2409] [Q&A]
Hi all, 

I have a problem regarding installing any python 3.6 - 3.10 on M1 Pro MacBook with x86_64 architecture.
I am following this [article](https://towardsdatascience.com/how-to-use-manage-multiple-python-versions-on-an-apple-silicon-m1-mac-d69ee6ed0250). 

- I have 2 different homebrew /usr and /opt
- same true for pyenv.
- I installed all dependencies via brew for both architecture
- this is my .zshrc. for pyenv and homebrew
```
# rosetta terminal setup
if [ $(arch) = "i386" ]; then
    alias brew86="/usr/local/bin/brew"
    alias pyenv86="/usr/local/bin/pyenv"

    # export CPPFLAGS="-I$(brew86 --prefix libffi)/include -I$(brew86 --prefix openssl)/include -I$(brew86 --prefix readline)/lib"
    # export CFLAGS="-I$(brew86 --prefix openssl)/include -I$(brew86 --prefix bzip2)/include -I$(brew86 --prefix readline)/include -I$(xcrun --show-sdk-path)/usr/include -Wno-implicit-function-declaration" 
    # export LDFLAGS="-L$(brew86 --prefix openssl)/lib -L$(brew86 --prefix readline)/lib -L$(brew86 --prefix zlib)/lib -L$(brew86 --prefix bzip2)/lib -L$(brew86 --prefix gettext)/lib -L$(brew86 --prefix libffi)/lib"
    # export PYTHON_BUILD_HOMEBREW_OPENSSL_FORMULA=openssl@1.0

fi

function intel-zsh(){
  arch -x86_64 /usr/local/bin/zsh
}

```

and my .zprofile 
```
export LANG="en_US.UTF-8"
export LC_ALL="en_US.UTF-8"
export LC_CTYPE="en_US.UTF-8"

eval "$(/opt/homebrew/bin/brew shellenv)"

export PYENV_ROOT="$HOME/.pyenv"
command -v pyenv >/dev/null || export PATH="$PYENV_ROOT/bin:$PATH"
eval "$(pyenv init -)"
```
Doesn't matter what compiler flags or versions I use, I keep getting
`BUILD FAILED (OS X 12.4 using python-build 20180424)`

by the way on native distribution everything works.

I would appreciate any help, If I overlooked some specific obvious settings, I am sorry in advance.

Br 

David

**Comment by native-api** [comment]
See https://github.com/pyenv/pyenv/issues/2324 and https://github.com/pyenv/pyenv/issues/1853
